### PR TITLE
Implement rename folder 1

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
This PR introduces the rename folder API for the GCS bucket interface, along with implementations in relevant structures. 
implementations will be addressed in a subsequent PR.
- Bucket handle
- Fast stat bucket
-  fake bucket 

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - Automated

### CODE REVIEW

File: internal/monitor/bucket.go [hunk 1]
```diff
@@ -203,6 +204,16 @@ func (mb *monitoringBucket) DeleteFolder(ctx context.Context, folderName string)
 	return err
 }
 
+func (mb *monitoringBucket) RenameFolder(
+	ctx context.Context,
```
Comment (vadlakondaswetha) [inline internal/monitor/bucket.go:208]: nit: Dont add each function parameter in new line:

go/go-style/decisions#func-formatting

Fix it everywhere.

File: internal/ratelimit/throttled_bucket.go [hunk 2]
```diff
@@ -198,14 +199,24 @@ func (b *throttledBucket) DeleteFolder(ctx context.Context, folderName string) (
 	return
 }
 
+func (b *throttledBucket) RenameFolder(
+	ctx context.Context,
+	folderName string,
+	destinationFolderId string) (o *control.RenameFolderOperation, err error) {
+	// Call through.
```
Comment (vadlakondaswetha) [inline internal/ratelimit/throttled_bucket.go:206]: Why there is no throttling here?

File: internal/storage/mock_bucket.go [hunk 3]
```diff
@@ -347,6 +348,36 @@ func (m *mockBucket) UpdateObject(p0 context.Context, p1 *gcs.UpdateObjectReques
 	return
 }
 
+func (m *mockBucket) RenameFolder(
+	ctx context.Context,
+	folderName string,
+	destinationFolderId string) (o0 *control.RenameFolderOperation, o1 error) {
+
```
Comment (vadlakondaswetha) [inline internal/storage/mock_bucket.go:355]: nit: remove this new line

File: internal/storage/mock_control_client.go [hunk 4]
```diff
@@ -44,6 +45,17 @@ func (m *MockStorageControlClient) DeleteFolder(ctx context.Context,
...
 }
 
+// Implement the RenameFolder method for the mock.
+func (m *MockStorageControlClient) RenameFolder(ctx context.Context,
+	req *controlpb.RenameFolderRequest,
+	opts ...gax.CallOption) (*control.RenameFolderOperation, error) {
+	args := m.Called(ctx, req, opts)
+	if folder, ok := args.Get(0).(*control.RenameFolderOperation); ok {
+		return folder, nil
+	}
```
Comment (vadlakondaswetha) [inline internal/storage/mock_control_client.go:55]: add new line after this.

**Review by vadlakondaswetha — COMMENTED** [review]
ADd in the description what is added in this PR and what is missed.

**Review by ankitaluthra1 — APPROVED** [review]
@Tulsishah Please add more granular description like which bucket types implementation is pending and which ones are completely done.

**Comment by Tulsishah** [comment]
> @Tulsishah Please add more granular description like which bucket types implementation is pending and which ones are completely done.

Missed fast stat bucket only, added that. Except that I think I have included all the things were there in description.

**Review by vadlakondaswetha — APPROVED** [review]

**Pull request merged**